src/scaffoldfitter/fitter.py

```diff
@@ -140,6 +142,10 @@ def decodeSettingsJSON(self, s: str, decoder):
         # ensure there is a first config step:
         if (len(self._fitterSteps) > 0) and isinstance(self._fitterSteps[0], FitterStepConfig):
             # field names are read (default to None), fields are found on load
+            id = settings.get("id")
```

Comment:
I think other tools specifying the structure of a JSON object are using 'identifier' and not 'id'.